packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -328,29 +220,170 @@ export default function BlockList( {
-				} )
-			}
-		</BlockListConsumer>
+			{ isRootList ? (
+				<BlockListProvider
+					value={ {
+						...DEFAULT_BLOCK_LIST_CONTEXT,
+						scrollRef: scrollViewRef.current,
+					} }
+				>
+					<BlockDraggableWrapper isRTL={ isRTL }>
+						{ ( { onScroll } ) => (
+							<KeyboardAwareFlatList
+								{ ...( Platform.OS === 'android'
+									? { removeClippedSubviews: false }
+									: {} ) } // Disable clipping on Android to fix focus losing. See https://github.com/wordpress-mobile/gutenberg-mobile/pull/741#issuecomment-472746541
+								accessibilityLabel="block-list"
+								innerRef={ ( ref ) => {
+									scrollViewRef.current = ref;
+								} }
+								extraScrollHeight={ extraScrollHeight }
+								keyboardShouldPersistTaps="always"
+								scrollViewStyle={ scrollViewStyle }
+								extraData={ getExtraData() }
+								scrollEnabled={ isRootList }
+								contentContainerStyle={ [
+									horizontal &&
+										styles.horizontalContentContainer,
+									isWider( blockWidth, 'medium' ) &&
+										( isContentStretch && isMultiBlocks
+											? styles.horizontalContentContainerStretch
+											: styles.horizontalContentContainerCenter ),
+								] }
+								style={ getStyles(
+									isRootList,
+									isStackedHorizontally,
+									horizontalAlignment
+								) }
+								data={ blockClientIds }
+								keyExtractor={ identity }
+								listKey={
+									rootClientId
+										? `list-${ rootClientId }`
+										: 'list-root'
+								}
+								renderItem={ renderItem }
+								CellRendererComponent={ BlockListItemCell }
+								shouldPreventAutomaticScroll={
+									shouldFlatListPreventAutomaticScroll
+								}
+								title={ title }
+								ListHeaderComponent={ header }
+								ListEmptyComponent={
+									! isReadOnly && (
+										<EmptyList
+											orientation={ orientation }
+											rootClientId={ rootClientId }
+											renderAppender={ renderAppender }
+											renderFooterAppender={
+												renderFooterAppender
+											}
+										/>
+									)
+								}
+								ListFooterComponent={
+									<Footer
+										addBlockToEndOfPost={
+											addBlockToEndOfPost
+										}
+										isReadOnly={ isReadOnly }
+										renderFooterAppender={
+											renderFooterAppender
+										}
+										withFooter={ withFooter }
+									/>
+								}
+								onScroll={ onScroll }
+							/>
+						) }
+					</BlockDraggableWrapper>
+				</BlockListProvider>
+			) : (
+				<>
+					{ blockClientIds.length > 0 ? (
+						<View
+							style={ [
+								getStyles(
+									isRootList,
+									isStackedHorizontally,
+									horizontalAlignment
+								),
+								horizontal && styles.horizontalContentContainer,
+								isWider( blockWidth, 'medium' ) &&
+									( isContentStretch && isMultiBlocks
+										? styles.horizontalContentContainerStretch
+										: styles.horizontalContentContainerCenter ),
+							] }
```

Comment:
These styles were applied in bulk to "fix" multi-column inner blocks. It needs further review, as all of the styles may not be necessary for inner blocks. Also, the Buttons block is not properly displaying Buttons inline, they erroneously wrap and overflow the parent container.